spec/modules.md

```diff
@@ -271,7 +273,11 @@ This algorithm takes a string `argument` and [configuration](#configuration)
   > This disallows circular `@use`s, which ensures that modules can't be used
   > until they're fully initialized.
 
-* Otherwise, return the result of [executing][] `file` with `config` and a new
+* Otherwise
+
+  * Keep track of which `config` was used for the `file` by its opaque ID.
```

Comment:
You don't need to explicitly specify when and where to track stuff in the spec—you can just refer to values and, as long as it's unambiguously specified what you're referring to, it's up to the implementation to figure out how to actually ensure the information is available when necessary.

In other words, specs only need to say _what to do_, they don't have to be concerned with _how to do it_.